python main.py config/gowalla/transformer.yml | tee term-outputs/train.txt
2024-10-30 10:08:32.672417: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-30 10:08:32.672502: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-30 10:08:32.674947: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-30 10:08:32.691621: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-30 10:08:34.117902: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
/home/ed/.local/lib/python3.9/site-packages/networkx/utils/backends.py:135: RuntimeWarning: networkx backend defined more than once: nx-loopback
  backends.update(_get_backends("networkx.backends"))
2024-10-30 10:08:41.551779: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-30 10:08:41.551840: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-30 10:08:41.553434: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-30 10:08:41.566576: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-30 10:08:42.971029: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
/home/ed/.local/lib/python3.9/site-packages/networkx/utils/backends.py:135: RuntimeWarning: networkx backend defined more than once: nx-loopback
  backends.update(_get_backends("networkx.backends"))
/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/transformer.py:286: UserWarning: enable_nested_tensor is True, but self.use_nested_tensor is False because encoder_layer.self_attn.batch_first was not
 True(use batch_first for better inference performance)                                                                                                                                                                      warnings.warn(f"enable_nested_tensor is True, but self.use_nested_tensor is False because {why_not_sparsity_fast_path}")
23 6 6
Total number of trainable parameters:  641685
terminate called after throwing an instance of 'c10::Error'
  what():  CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Exception raised from c10_cuda_check_implementation at ../c10/cuda/CUDAException.cpp:44 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7efef64ced87 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x7efef647f75f in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10.so)
frame #2: c10::cuda::c10_cuda_check_implementation(int, char const*, char const*, int, bool) + 0x118 (0x7efef659f8a8 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10_cuda.so)
frame #3: <unknown function> + 0x1d40e (0x7efef656a40e in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10_cuda.so)
frame #4: <unknown function> + 0x1f744 (0x7efef656c744 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10_cuda.so)
frame #5: <unknown function> + 0x1fb6d (0x7efef656cb6d in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10_cuda.so)
frame #6: <unknown function> + 0x540080 (0x7efef5340080 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libtorch_python.so)
frame #7: <unknown function> + 0x649bf (0x7efef64b39bf in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10.so)
frame #8: c10::TensorImpl::~TensorImpl() + 0x21b (0x7efef64acc8b in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10.so)
frame #9: c10::TensorImpl::~TensorImpl() + 0x9 (0x7efef64ace39 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libc10.so)
frame #10: <unknown function> + 0x802058 (0x7efef5602058 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libtorch_python.so)
frame #11: THPVariable_subclass_dealloc(_object*) + 0x2f6 (0x7efef56023d6 in /home/ed/.local/lib/python3.9/site-packages/torch/lib/libtorch_python.so)
frame #12: <unknown function> + 0x127ab8 (0x557f84da7ab8 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #13: <unknown function> + 0x127ba5 (0x557f84da7ba5 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #14: <unknown function> + 0x14ff14 (0x557f84dcff14 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #15: <unknown function> + 0x1344c8 (0x557f84db44c8 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #16: <unknown function> + 0x1480be (0x557f84dc80be in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #17: <unknown function> + 0x1480ab (0x557f84dc80ab in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #18: <unknown function> + 0x1480ab (0x557f84dc80ab in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #19: <unknown function> + 0x147cbb (0x557f84dc7cbb in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #20: <unknown function> + 0x1227a8 (0x557f84da27a8 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #21: <unknown function> + 0x21a4a3 (0x557f84e9a4a3 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #22: _PyEval_EvalFrameDefault + 0x4a6f (0x557f84db00ef in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #23: <unknown function> + 0x12a6a7 (0x557f84daa6a7 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #24: _PyFunction_Vectorcall + 0xb9 (0x557f84dbca99 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #25: _PyEval_EvalFrameDefault + 0x669 (0x557f84dabce9 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #26: <unknown function> + 0x13cda3 (0x557f84dbcda3 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #27: _PyEval_EvalFrameDefault + 0x3be (0x557f84daba3e in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #28: <unknown function> + 0x12a6a7 (0x557f84daa6a7 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #29: _PyFunction_Vectorcall + 0xb9 (0x557f84dbca99 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #30: _PyEval_EvalFrameDefault + 0x11db (0x557f84dac85b in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #31: <unknown function> + 0x12a6a7 (0x557f84daa6a7 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #32: _PyEval_EvalCodeWithName + 0x47 (0x557f84daa367 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #33: PyEval_EvalCodeEx + 0x39 (0x557f84daa319 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #34: PyEval_EvalCode + 0x1b (0x557f84e6566b in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #35: <unknown function> + 0x213c89 (0x557f84e93c89 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #36: <unknown function> + 0x20fe94 (0x557f84e8fe94 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #37: PyRun_StringFlags + 0x9d (0x557f84e87b2d in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #38: PyRun_SimpleStringFlags + 0x3d (0x557f84e87a5d in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #39: Py_RunMain + 0x262 (0x557f84e86c32 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #40: Py_BytesMain + 0x39 (0x557f84e593a9 in /home/ed/miniconda3/envs/MHSA-env/bin/python)
frame #41: <unknown function> + 0x29590 (0x7efef7229590 in /lib64/libc.so.6)
frame #42: __libc_start_main + 0x80 (0x7efef7229640 in /lib64/libc.so.6)
frame #43: <unknown function> + 0x1d92b1 (0x557f84e592b1 in /home/ed/miniconda3/envs/MHSA-env/bin/python)

Traceback (most recent call last):
  File "/home/ed/ppdp-project/MHSA/main.py", line 132, in <module>
    mp.spawn(
  File "/home/ed/.local/lib/python3.9/site-packages/torch/multiprocessing/spawn.py", line 241, in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method="spawn")
  File "/home/ed/.local/lib/python3.9/site-packages/torch/multiprocessing/spawn.py", line 197, in start_processes
    while not context.join():
  File "/home/ed/.local/lib/python3.9/site-packages/torch/multiprocessing/spawn.py", line 158, in join
    raise ProcessRaisedException(msg, error_index, failed_process.pid)
torch.multiprocessing.spawn.ProcessRaisedException: 

-- Process 0 terminated with the following error:
Traceback (most recent call last):
  File "/home/ed/.local/lib/python3.9/site-packages/torch/multiprocessing/spawn.py", line 68, in _wrap
    fn(i, *args)
  File "/home/ed/ppdp-project/MHSA/main.py", line 73, in main
    res_single = single_run(train_loader, val_loader, test_loader, config, rank, log_dir)
  File "/home/ed/ppdp-project/MHSA/main.py", line 26, in single_run
    model, perf = get_trainedNets(config, model, train_loader, val_loader, device, log_dir)
  File "/home/ed/ppdp-project/MHSA/utils/utils.py", line 47, in get_trainedNets
    best_model, performance = trainNet(config, model, train_loader, val_loader, device, log_dir=log_dir)
  File "/home/ed/ppdp-project/MHSA/utils/train.py", line 187, in trainNet
    globaliter = train(
  File "/home/ed/ppdp-project/MHSA/utils/train.py", line 267, in train
    logits = model(x, x_dict, device)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1523, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1359, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ed/ppdp-project/MHSA/models/MHSA.py", line 35, in forward
    emb = self.Embedding(src, context_dict)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ed/ppdp-project/MHSA/models/embed.py", line 152, in forward
    emb = emb + self.temporal_embedding(context_dict["time"], context_dict["weekday"])
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ed/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ed/ppdp-project/MHSA/models/embed.py", line 46, in forward
    hour = torch.div(time, self.minute_size, rounding_mode="floor")
RuntimeError: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.


